src/stave.js

```diff
@@ -319,14 +298,96 @@ Vex.Flow.Stave = (function() {
 
     addEndTimeSignature: function(timeSpec, customPadding) {
       this.addEndModifier(new Vex.Flow.TimeSignature(timeSpec, customPadding));
+      return this;
     },
 
     addTrebleGlyph: function() {
-      this.clef = "treble";
-      this.addGlyph(new Vex.Flow.Glyph("v83", 40));
+      this.addClef('treble');
       return this;
     },
 
+    getModifiers: function(position) {
+      if (position === undefined) return this.modifiers;
+
+      return this.modifiers.filter(function(modifier) {
+        return modifier.getPosition() === position;
+      });
+    },
+
+    sortByCategory: function(items, order) {
+      for (var i = items.length - 1; i >= 0; i--) {
+        for (var j = 0; j < i; j++) {
+          if (order[items[j].getCategory()] > order[items[j + 1].getCategory()]) {
+            var temp = items[j];
+            items[j] = items[j + 1];
+            items[j + 1] = temp;
+          }
+        }
+      }
+    },
+
+    format: function() {
```

Comment:
Allowing only one stavemodifier for each category + position combination will certainly have affect to current tests, but that test examples (adding multiple time signature, key or clef to a single stave) are not used in real world so I think most of existing users will not experience any change. 

And there's another good thing, users will be able to change existing stave modifiers(clef, time or key) although they already added clef, time or key signature.

Even if there exists users who want to draw multiple clefs(or etc) in a row in the single stave, they can do by using clefnote of timesignote.

Hmm... may be we can keep both addClef..+ and setClef...+
It might be better to only simply setting deprecated flag comments to that add functions.
How do you think about current add functions? (addClef, addEndClef, addTimeSignature, addEndTimeSignature and addKeySignature)